test/parallel/test-child-process-flush-stdio.js

```diff
@@ -20,10 +21,42 @@ function spawnWithReadable() {
     assert.strictEqual(code, 0);
     assert.strictEqual(signal, null);
     assert.strictEqual(Buffer.concat(buffer).toString().trim(), '123');
+    spawnWithPipe();
   }));
   p.stdout.on('readable', function() {
-    let buf;
-    while (buf = this.read())
-      buffer.push(buf);
+    setImmediate(() => {
+      let buf;
+      while (buf = this.read())
+        buffer.push(buf);
+    });
   });
 }
+
+function spawnWithPipe() {
+  const buffer = [];
+  const through = new stream.PassThrough();
+  const p = cp.spawn('seq', [ '36000' ]);
```

Comment:
Agreed. I can PR just the testcase separately although I'm not sure if it adds any value since the issue turned out to be in the stream implementation.